smaps/src/main/java/com/google/smaps/FileParser.java

```diff
@@ -57,14 +57,31 @@
     // TODO(@sophbohr22): Add a defense to the parser so that files with extremely large region
     // counts aren't allowed.
 
-    // Indicates whether a new region is being evaluated at each iteration.
-    boolean nextRegion = true;
+    // Indicates whether there is a previous region to be built (will only be false for the first
+    // region).
+    boolean prevRegion = false;
 
+    // Initialize lineNumber, and loop through the file while there's still a new line.
     int lineNumber = 0;
     while (sc.hasNextLine()) {
       lineNumber++;
       String line = sc.nextLine();
-      if (nextRegion) {
+      if (checkNewRegion(line)) {
+        // Build the previous region (if there is one) and add it to the list.
+        if (prevRegion) {
+          // Ensure that a valid size was set for this region.
+          Region r = region.build();
+          if (r.size() == -1) {
+            session.setAttribute("fileErrorMessage",
+                "One or more regions does not contain a 'size' field, see go/smaps-vis/smaps-example.txt to view properly formatted smaps file.");
```

Comment:
We shouldn't direct users to a go link in either of these errors, it won't work for all users.
I think just a simple "Required 'size' field not found in smaps file." or something similar would be sufficient.